File: 11vaFeriaAcademicayCultural/Basico/Problema6/problema6.py
"""Problema 6 Lenguaje de la F
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
Python 2.7"""

from sys import stdin, stdout

if __name__ == '__main__':
    #  el metodo readline de stdin lee la siguiente linea de la entrada
    #  incluyendo el salto de linea, para eliminarlo usamos el metodo strip (sin parametros)
    #  de la clase str para eliminar los caracteres no deseados (espacios y saltos de linea)
    accion = stdin.readline().strip()
    mensaje = stdin.readline().strip()
    vocales = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'} #  conjunto para identificar las vocales
    nuevo_mensaje = [] #  lista para crear el nuevo mensaje
    ignorar = 0
    for letra in mensaje:
        if not ignorar: #  si ignorar es mayor a cero (en Python, el valor booleano de 0 es False)
            nuevo_mensaje.append(letra)
            if letra in vocales: #  realizamos una busqueda de la letra entre las vocales, el cual es
                                 #  letra en la cual se basa el lenguaje de la F
                if accion == 'E':
                    """Si la accion es encriptar, debemos de agregar en la lista despues de la vocal la letra f, seguido de la
                    vocal encontrada en minuscula"""
                    nuevo_mensaje.extend(['f', letra.lower()])
                else:
                    """Si la accion es desencriptar, debemos de ignorar las siguientes 2 letras despues de la vocal"""
                    ignorar = 2
        else:
            ignorar-=1 #  restamos en 1 a cada iteracion para solo ignorar 2 letras

    stdout.write('{}\n'.format(''.join(nuevo_mensaje)))

File: 11vaFeriaAcademicayCultural/Basico/Problema6/problema6.java
/*Problema 5 Buscando un caracter
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
Java*/

import java.util.Scanner;
import java.util.ArrayList;

class Problema6{
  public static void main(String[] args) {
    Scanner entrada = new Scanner(System.in);
    char accion = entrada.nextLine().charAt(0); // declaramos como char al caracter buscado
    String mensaje = entrada.nextLine();
    int ignorar = 0;
    ArrayList<Character> nuevoMensaje = new ArrayList<Character>(); // implementamos un ArrayList para almacenar los caracteres
                                                                    //del nuevo mensaje
    for(int i = 0; i < mensaje.length(); i++){//for para imprimir la parte superior de la piramide
      if(ignorar == 0){
        char letra = mensaje.charAt(i);
        nuevoMensaje.add(letra);
        if(esVocal(letra)){
          if(accion == 'E'){
            /*Si la accion es encriptar, debemos de agregar en la lista despues de la vocal la letra f, seguido de la
            vocal encontrada en minuscula*/
            nuevoMensaje.add('f');
            nuevoMensaje.add(Character.toLowerCase(letra));
          }
          else{
            ignorar = 2; //Si la accion es desencriptar, debemos de ignorar las siguientes 2 letras despues de la vocal
          }
        }
      }
      else{ignorar--;}
    }
    for(int j = 0; j < nuevoMensaje.size(); j++){System.out.print(nuevoMensaje.get(j));}
    System.out.println();
  }
  public static boolean esVocal(char letra){
    //funcion para ve
    char vocales[] = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}; //arreglo para las vocales
    for (int i = 0; i < 10; i++){if(letra == vocales[i]){return true;}}
    return false;
  }
}
